src/System.IO.Compression.Brotli/System/IO/Compression/BrotliStream.cs

```diff
@@ -346,27 +310,19 @@ public override void Write(byte[] buffer, int offset, int count)
                     throw new TimeoutException(BrotliEx.TimeoutWrite);
                 }
                 copyLen = bytesRemain > _bufferSize ? _bufferSize : bytesRemain;
-                Marshal.Copy(buffer, currentOffset, _bufferInput, copyLen);
+                byte[] _bufferInput = new byte[copyLen];
```

Comment:
@shiftylogic @ahsonkhan Also I need an advice about this variable. I now think about remove this bufferInput and just use buffer instead, but in this case offset and count make no sense.